### ISSUE TITLE
Move out older versions of docs to a new store

### ISSUE THREAD

**Issue by pankajkoti** [apache/airflow-site#773]
We recently ran out of space for the documentation build and we have a thread mentioning the same here https://lists.apache.org/thread/gh17psdn39s1o05lwxfqvvn5htjtqs05

Find out a way to move older versions of the documentation to a new place and redirect links to that store.

**Comment by pankajkoti** [comment]
We're exploring an approach to move older versions of the docs to a newly created https://github.com/apache/airflow-site-archive repository and will keep it in the github-pages branch. We need to figure out how to handle the redirects back and forth with respect to choosing the version from the drop-down.

**Comment by pankajkoti** [comment]
When I do a `./site.sh build-site` after following the steps outlined in README.md and installing `brew install hugo`, I get the below error and currently stuck at resolving this

```
WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets:
  chunk-4.3d5f5.js (1.53 MiB)
$ cross-env HUGO_ENV=production hugo -d ../dist -s site -v
Start building sites …
hugo v0.111.3+extended darwin/arm64 BuildDate=unknown
INFO 2023/04/26 19:02:52 syncing static files to /
ERROR 2023/04/26 19:02:54 render of "page" failed: "/Users/pankajkoti/airflow-site/landing-pages/site/layouts/_default/baseof.html:23:7": execute of template failed: template: _default/search.html:23:7: executing "_default/search.html" at <partial "head.html" .>: error calling partial: execute of template failed: html/template:partials/head.html:15:17: no such template "_internal/google_news.html"
ERROR 2023/04/26 19:02:54 render of "page" failed: "/Users/pankajkoti/airflow-site/landing-pages/site/layouts/blog/baseof.html:23:7": execute of template failed: template: blog/single.html:23:7: executing "blog/single.html" at <partial "head.html" .>: error calling partial: execute of template failed: html/template:partials/head.html:15:17: no such template "_internal/google_news.html"
ERROR 2023/04/26 19:02:54 render of "page" failed: "/Users/pankajkoti/airflow-site/landing-pages/site/layouts/blog/baseof.html:23:7": execute of template failed: template: blog/single.html:23:7: executing "blog/single.html" at <partial "head.html" .>: error calling partial: execute of template failed: html/template:partials/head.html:15:17: no such template "_internal/google_news.html"
ERROR 2023/04/26 19:02:54 render of "page" failed: "/Users/pankajkoti/airflow-site/landing-pages/site/layouts/blog/baseof.html:23:7": execute of template failed: template: blog/single.html:23:7: executing "blog/single.html" at <partial "head.html" .>: error calling partial: execute of template failed: html/template:partials/head.html:15:17: no such template "_internal/google_news.html"
Error: Error building site: failed to render pages: render of "page" failed: "/Users/pankajkoti/airflow-site/landing-pages/site/layouts/blog/baseof.html:23:7": execute of template failed: template: blog/single.html:23:7: executing "blog/single.html" at <partial "head.html" .>: error calling partial: execute of template failed: html/template:partials/head.html:15:17: no such template "_internal/google_news.html"
Total in 2060 ms
error Command failed with exit code 255.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 255.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by potiuk** [comment]
I think this is about bad hugo version - look at the CI of ours, I think I had very similar issue when I tried to build the docs on Mac - and I could not solve it when I tried. One of the ways of solving it was to use a docker container based on debian to build the docs. You **could** potentially use Breeze image for it (if you for example check-out the sites in "files" folder) but it might have another problem it could suffer from - slow filesystem mounted to docker on Mac,

Going Linux/Debian first (maybe using remote build machine for it) is likely the fastest way to solve the problem.

There is also an idea to modernise our build pipeline for the sites  - wich could solve the problem.

**Comment by pankajkoti** [comment]
yes, looks like the template was removed https://discourse.gohugo.io/t/page-render-error/43594

**Comment by pankajkoti** [comment]
Our CI seems to user Hugo version` 0.91.2` as per https://github.com/apache/airflow-site/actions/runs/4253003040/jobs/7397288933, so I am installing that version now with
`go install -tags extended github.com/gohugoio/hugo@v0.91.2` and see how it goes thereafter.

**Comment by pankajkoti** [comment]
That helped @potiuk , thank you! The site was built and I can load the index page, however, the docs are not loading. Do I need to do something additional while building?
<img width="1728" alt="Screenshot 2023-04-26 at 7 37 56 PM" src="https://user-images.githubusercontent.com/10206082/234602257-ae2ca313-e5da-42f4-95ac-f108342801cc.png">


<img width="1719" alt="Screenshot 2023-04-26 at 7 38 06 PM" src="https://user-images.githubusercontent.com/10206082/234602298-82259039-a62a-4611-bcbf-cad775b78b12.png">

cc: @jedcunningham @phanikumv

**Comment by potiuk** [comment]
I do **not** know the system that well, I think at least in some cases building the docs history has been skipped to save the size of generated images - look at the CI steps, I think there was even a comment about it

**Comment by ashb** [comment]
Did we talk about keeping all the old versions somewhere else than the main branch (separated detached branches in this repo?)

**Comment by pankajkoti** [comment]
> Since build-site is copying stuff from docs-archive into dist, and we upload dist, can we not just delete docs-archive once we are done building? That'll give us 10gb of extra room without having any negatives or extra complexity?

Yes, @jedcunningham. I tried your suggestion and have created a draft [PR](https://github.com/apache/airflow-site/pull/777).

I pushed 2 commits to display disk-free CLI command (`df -h`) output after each of the significant steps in our CI job ([bf25b69](https://github.com/apache/airflow-site/pull/777/commits/bf25b69fa3c43c0088c2da7534de6152ac452ba4) with the existing setup just adding a `df -h` and [6bc8ab7](https://github.com/apache/airflow-site/pull/777/commits/6bc8ab718ef1410c5486b1cafdc473e9aa3ce17f) after removing `docs-archive` directory to display `df -h` output).
CI job with first commit - https://github.com/apache/airflow-site/actions/runs/4830163412/jobs/8606043658?pr=777
CI job with second commit  - https://github.com/apache/airflow-site/actions/runs/4830346989/jobs/8606461864?pr=777

I did a search in the repo and found there is no other reference to `docs-archive` after the site is built, so I believe we're safe to remove it from the CI job for the subsequent steps once the site is built. This step kept before the `Deploy website on asf-site branch` step will ensure (🤞🏽 due to +10GB reclaimed) that CI build will not fail as observed in this step last time due to no disk space available.

I am attaching the outputs (both PDF and PNG; you will need to zoom in unfortunately since I did a full-page capture) of the CI job with the 2 commits mentioned above.

PDF output:
first commit: [before_pdf.pdf](https://github.com/apache/airflow-site/files/11353528/before_pdf.pdf)
second commit: [after_pdf.pdf](https://github.com/apache/airflow-site/files/11353526/after_pdf.pdf)

PNF Output:
first commit: 
![Before](https://user-images.githubusercontent.com/10206082/235149518-b7db7a43-c375-4802-bf2a-a512ae056a80.png)
second commit: 
![after](https://user-images.githubusercontent.com/10206082/235149615-31c9ed6e-be3a-4b80-877f-a76a2cfa535b.png)



If this solution is accepted, I guess this is a quick win for us for now :)

**Comment by pankajkoti** [comment]
> Did we talk about keeping all the old versions somewhere else than the main branch (separated detached branches in this repo?)

yes Ash, thank you for your comment. Jed had suggested this idea too and then we decided to try the other repository approach first. I am sorry I don't remember the reasoning but maybe @jedcunningham can tell more about his thought process.

**Comment by jedcunningham** [comment]
Yeah, that's more or less what I was thinking, but I hadn't quite connected the dots. The good news is I don't think we need to worry about any extra complexity now 🍺.

**Comment by potiuk** [comment]
Yeah. I like it too, though it would be great to modernize things a bit as well :)  - but I agree if we can NOT involve another repo/redirection we seen ti ve good for now. 

What we can do though - we could potentially rewrite the history for the whole airflow-site and keep maybe few last commits ? And we could do it periodically.  That would leave us even more space I think and all the operations on the repo would be much faster (for now just getting my liquidprompt to show the version takes visible time.

**Comment by jedcunningham** [comment]
I'd be 100% on board if squashing it all helps. It's become really painful!

**Comment by pankajkoti** [comment]
PR https://github.com/apache/airflow-site/pull/777 was merged which reclaims us an additional 10GB+ space for the failing CI build job step, we do not need to move out the docs to the newer repo in the near future. Closing it for now

@potiuk  Can we keep the new repo or do we need to archive `airflow-site-archive`?
